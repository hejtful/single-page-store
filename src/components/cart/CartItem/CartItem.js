import React from 'react';

import styles from './CartItem.module.css';

export const CartItem = ({ id, title, price, quantity, onRemove }) => {
  function handleRemove() {
    onRemove(id);
  }

  return (
    <div className={styles.wrapper}>
      <div className={styles.infoWrapper}>
        <div className={styles.line}>
          <h4 className={styles.title}>{title}</h4>
          <span>{price}</span>
        </div>

        <div className={styles.line}>
          <span>Quantity:</span>
          <span>{quantity}</span>
        </div>
      </div>

      <div className={styles.removeWrapper}>
        <button type="button" onClick={handleRemove}>
          X
        </button>
      </div>
    </div>
  );
};
