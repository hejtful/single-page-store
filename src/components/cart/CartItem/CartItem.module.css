.wrapper {
  align-items: center;
  display: flex;
  margin-top: 1em;
  padding-bottom: 0.5em;
}

.infoWrapper {
  border-bottom: 1px solid #363636;
  flex-grow: 1;
}

.line {
  display: flex;
  justify-content: space-between;
  margin-bottom: 0.5em;
}

.title {
  margin: 0;
}

.removeWrapper {
  flex: 0 0 1em;
  margin-left: 1em;
}
