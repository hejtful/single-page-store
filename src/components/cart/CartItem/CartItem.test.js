import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { CartItem } from './CartItem';

const props = {
  id: '1',
  title: 'Test title',
  price: '299.99',
  quantity: 5,
};

const methods = {
  onRemove: jest.fn(),
};

it('renders with props and remove button', () => {
  const { container } = render(<CartItem {...props} {...methods} />);

  expect(container).toMatchSnapshot();
});

it('should call onRemove with item ID when remove button is clicked', () => {
  const { getByRole } = render(<CartItem {...props} {...methods} />);

  userEvent.click(getByRole('button'));

  expect(methods.onRemove).toBeCalledWith(props.id);
});
