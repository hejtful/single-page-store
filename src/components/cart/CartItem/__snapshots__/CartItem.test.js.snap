// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with props and remove button 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      class="infoWrapper"
    >
      <div
        class="line"
      >
        <h4
          class="title"
        >
          Test title
        </h4>
        <span>
          299.99
          $
        </span>
      </div>
      <div
        class="line"
      >
        <span>
          Quantity:
        </span>
        <span>
          5
        </span>
      </div>
    </div>
    <div
      class="removeWrapper"
    >
      <button
        class="danger"
        type="button"
      >
        X
      </button>
    </div>
  </div>
</div>
`;
